kotlin-asyncapi-annotation/src/main/kotlin/org/openfolder/kotlinasyncapi/annotation/Schema.kt

```diff
@@ -2,34 +2,13 @@ package org.openfolder.kotlinasyncapi.annotation
 
 import kotlin.reflect.KClass
 
+@Target(
+    AnnotationTarget.CLASS,
+    AnnotationTarget.ANNOTATION_CLASS
+)
+@Repeatable
+@AsyncApiAnnotation
 annotation class Schema(
     val default: Boolean = false,
-    val reference: String = "",
-    val title: String = "",
-    val description: String = "",
-    val implementation: KClass<*> = Void::class,
-    val readOnly: Boolean = false,
-    val writeOnly: Boolean = false,
-    val examples: Array<String> = [],
-    val type: String = "",
-    val multipleOf: Int = 0,
-    val maximum: Int = 0,
-    val exclusiveMaximum: Int = 0,
-    val minimum: Int = 0,
-    val exclusiveMinimum: Int = 0,
-    val maxLength: Int = 0,
-    val minLength: Int = 0,
-    val pattern: String = "",
-    val maxProperties: Int = 0,
-    val minProperties: Int = 0,
-    val required: Array<String> = [],
-    val additionalProperties: String = "",
-    val allOf: Array<KClass<*>> = [],
-    val anyOf: Array<KClass<*>> = [],
-    val oneOf: Array<KClass<*>> = [],
-    val not: KClass<*> = Void::class,
-    val format: String = "",
-    val discriminator: String = "",
-    val externalDocs: ExternalDocumentation = ExternalDocumentation(default = true, url = ""),
-    val deprecated: Boolean = false
+    val implementation: KClass<*> = Void::class
```

Comment:
Why did we remove all of the schema fields? How is the `implementation` field replacing this?

Reply:
we currently would ignore all other fields because we only use the implementation field to generate the schema object from it. we would need to implement an extension logic to also include the other field, which is not feasible for now.